fe/fe-core/src/main/java/org/apache/doris/datasource/iceberg/source/IcebergScanNode.java

```diff
@@ -212,60 +212,73 @@ private List<Split> doGetSplits() throws UserException {
+                    String dataFilePath = normalizeLocation(splitTask.file().path().toString());
+
+                    List<String> partitionValues = new ArrayList<>();
+                    if (isPartitionedTable) {
+                        StructLike structLike = splitTask.file().partition();
+                        List<PartitionField> fields = splitTask.spec().fields();
+                        Types.StructType structType = icebergTable.schema().asStruct();
+
+                        // set partitionValue for this IcebergSplit
+                        for (int i = 0; i < structLike.size(); i++) {
+                            Object obj = structLike.get(i, Object.class);
+                            String value = String.valueOf(obj);
+                            PartitionField partitionField = fields.get(i);
+                            if (partitionField.transform().isIdentity()) {
+                                Type type = structType.fieldType(partitionField.name());
+                                if (type != null && type.typeId().equals(Type.TypeID.DATE)) {
+                                    // iceberg use integer to store date,
+                                    // we need transform it to string
+                                    value = DateTimeUtil.daysToIsoDate((Integer) obj);
+                                }
                             }
+                            partitionValues.add(value);
                         }
-                        partitionValues.add(value);
+
+                        // Counts the number of partitions read
+                        partitionPathSet.add(structLike.toString());
+                    }
+                    LocationPath locationPath = new LocationPath(dataFilePath, source.getCatalog().getProperties());
+                    Path finalDataFilePath = locationPath.toStorageLocation();
+                    IcebergSplit split = new IcebergSplit(
+                            finalDataFilePath,
+                            splitTask.start(),
+                            splitTask.length(),
+                            splitTask.file().fileSizeInBytes(),
+                            new String[0],
+                            formatVersion,
+                            source.getCatalog().getProperties(),
+                            partitionValues);
+                    if (formatVersion >= MIN_DELETE_FILE_SUPPORT_VERSION) {
+                        split.setDeleteFileFilters(getDeleteFileFilters(splitTask));
                     }
+                    split.setTableFormatType(TableFormatType.ICEBERG);
+                    splits.add(split);
 
-                    // Counts the number of partitions read
-                    partitionPathSet.add(structLike.toString());
+                    // End loop early as one split is enough if the statement can push down count
+                    if (canPushCount) {
+                        break;
```

Comment:
This is what i want to do. End the entire loop early to avoid a lot of useless spilts if the statement can push down count.